File: Data Structures & Algorithms/longest-common-prefix/submission-4.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string}
     */
    longestCommonPrefix(strs) {
        if(strs.length == 0) {
            return "";
        }

        for(let i = 0; i < strs[0].length; i++) {
            let ch = strs[0][i];
            for(let s of strs) {
                if(i >= s.length || s[i] != ch) {
                    return s.substring(0, i);
                }
            }
        }
        return strs[0];
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-4.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map <String, List<String>> ans = new HashMap<>();

        for (String word: strs) {
            int[] count = new int[26];
            for (char l : word.toCharArray()) {
                count[l - 'a']++;
            }
            String key = Arrays.toString(count);
            if(!ans.containsKey(key)) {
                ans.put(key, new ArrayList<>());
            }
            ans.get(key).add(word);
        }
        return new ArrayList<>(ans.values());

    }
}
